apps/findmydevice/js/findmydevice.js

```diff
@@ -117,30 +155,32 @@ var FindMyDevice = {
   },
 
   _initSettings: function fmd_init_settings(callback) {
-    // for each setting we're interested in, create a SettingsHelper,
-    // cache its initial value in a property (e.g., _enabled), and
-    // store the SettingsHelper as another property (e.g., _enabledHelper).
+    // for each setting we're interested in, create a LockedSettingsHelper,
+    // cache its initial value in a property (e.g., _enabled), and store the
+    // LockedSettingsHelper as another property (e.g., _enabledHelper).
     var settingsToProperties = {
       // setting name : property name for cached value
       'findmydevice.enabled': '_enabled',
       'findmydevice.registered': '_registered',
       'findmydevice.current-clientid': '_currentClientID',
-      'findmydevice.can-disable': '_canDisable'
+      'findmydevice.can-disable': '_canDisable',
+      'findmydevice.retry-count': '_retryCount'
     };
     var settings = Object.keys(settingsToProperties);
 
-    var loaded = 0;
-    var totalSettings = Object.keys(settingsToProperties).length;
-    settings.forEach(function(s) {
+    return Promise.all(settings.map((s, i) => {
```

Comment:
Can we have something easier to read ? Why do we need 'i' here ?
Object.keys(settingsToProperties).map( (X, Y) => ) with X and Y with a name.